classlib/src/main/java/org/teavm/classlib/java/util/TArrays.java

```diff
@@ -37,84 +39,90 @@
 
 public class TArrays extends TObject {
     public static char[] copyOf(char[] array, int length) {
-        char[] result = new char[length];
         int sz = TMath.min(length, array.length);
-        for (int i = 0; i < sz; ++i) {
-            result[i] = array[i];
+        if (length < 0) {
```

Comment:
This shoud not be done here. Line below instantiates array and JVM should throw exception there.

Reply:
JVM don't throw required exception. I investigated create array instructions and they indeed just create array without any check (and we get error in JS). Still, I don't know how to add preliminary check and throw. Maybe some example?

Reply:
https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-6.html#jvms-6.5.anewarray

Reply:
https://github.com/konsoletyper/teavm/blob/master/core/src/main/java/org/teavm/parsing/ProgramParser.java#L478 here this anewarray is converted to instruction. But I don't see in code where length checks and throws are added then.